python/tvm/topi/hexagon/slice_ops/qadd_qsub_qmul.py

```diff
@@ -0,0 +1,253 @@
+                    + (tensor_B[n * n_b, h * h_b, w * w_b, c * c_b] * scale_b)
+                    + corr
+                )
+                >> rsh
+            ),
+            "uint8",
+        ).astype("uint8"),
+    )
+
+    """
+    
+    return te.compute(
+        output_shape,
+        lambda n, h, w, c: (((scale_A/scale_M) * (tensor_A[n * n_a, h * h_a, w * w_a, c * c_a] - zero_point_A))
+        + ((scale_B/scale_M) * (tensor_B[n * n_b, h * h_b, w * w_b, c * c_b] - zero_point_B))
+        + zero_point_M).astype('uint8'),
+    )
+    """
+
+
+def qsubtract_broadcast_compute(
+    tensor_A,
+    tensor_B,
+    output_shape,
+    zero_point_A,
+    scale_A,
+    zero_point_B,
+    scale_B,
+    zero_point_M,
+    scale_M,
+):
+    A_broadcast, B_broadcast = broadcast_axis(tensor_A, tensor_B)
+    n_a, h_a, w_a, c_a = A_broadcast
+    n_b, h_b, w_b, c_b = B_broadcast
+
+    scale_a, scale_b, rsh, corr = get_int_scale(
+        scale_A, scale_B, scale_M, zero_point_A, zero_point_B, zero_point_M, "qsub"
+    )
+
+    return te.compute(
+        output_shape,
+        lambda n, h, w, c: saturate(
+            (
+                (
+                    (tensor_A[n * n_a, h * h_a, w * w_a, c * c_a] * scale_a)
+                    - (tensor_B[n * n_b, h * h_b, w * w_b, c * c_b] * scale_b)
+                    + corr
+                )
+                >> rsh
+            ),
+            "uint8",
+        ).astype("uint8"),
+    )
+
+
+def qmultiply_broadcast_compute(
+    tensor_A,
+    tensor_B,
+    output_shape,
+    zero_point_A,
+    scale_A,
+    zero_point_B,
+    scale_B,
+    zero_point_M,
+    scale_M,
+):
+    A_broadcast, B_broadcast = broadcast_axis(tensor_A, tensor_B)
+    n_a, h_a, w_a, c_a = A_broadcast
+    n_b, h_b, w_b, c_b = B_broadcast
+
+    scale_int, rsh, corr = get_int_scale(
+        scale_A, scale_B, scale_M, zero_point_A, zero_point_B, zero_point_M, "qmul"
+    )
+
+    return te.compute(
+        output_shape,
+        lambda n, h, w, c: saturate(
+            (
+                (
+                    scale_int
+                    * (tensor_A[n * n_a, h * h_a, w * w_a, c * c_a] - zero_point_A)
+                    * (tensor_B[n * n_b, h * h_b, w * w_b, c * c_b] - zero_point_B)
+                    + corr
+                )
+                >> rsh
+            ),
+            "uint8",
+        ).astype("uint8"),
+    )
+
+
+def tir_schedule(
+    out_m,
+    tensor_A,
+    tensor_B,
+    output_layout: str,
+    tensor_A_layout: str,
+    tensor_B_layout: str,
+    op_name: str,
+):
+    """Schedule for input and output layout nhwc-8h8w32c-2d"""
+    func = te.create_prim_func([tensor_A, tensor_B, out_m])
+
+    print(func)
+
+    s = tir.Schedule(func)
+
+    block = s.get_block("compute")
+
+    if tensor_A_layout == "nhwc-8h8w32c-2d":
+        tensor_A_transformed_layout = get_layout_transform_fn(tensor_A_layout)
+        s.transform_layout(block, buffer=("read", 0), index_map=tensor_A_transformed_layout)

```

Comment:
I believe the error is caught here. buffer=("read", 0) should return tensor_A and that has the layout "nhwc-8h8w32c-2d" so we should be able to apply transform_layout() for this buffer. But buffer=("read", 0) is returning tensor_B (shape - [1, 8, 8, 1]) in this case and since it is not a multiple of crouton, the code fails